src/lsp_data.rs

```diff
@@ -87,72 +90,56 @@ impl Location {
             uri: Url::from_file_path(&file_name).unwrap().into_string(),
             range: Range {
                 start: Position {
-                    line: line,
-                    character: col,
+                    line: from_usize(line),
+                    character: from_usize(col),
                 },
                 end: Position {
-                    line: line,
-                    character: col,
+                    line: from_usize(line),
+                    character: from_usize(col),
                 },
             },
         }
     }
 }
 
-/// The initialize request is sent as the first request from the client to the server.
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct InitializeParams {
-    pub processId: usize,
-    pub rootPath: PathBuf
-}
-
-#[derive(Debug, Deserialize)]
-pub struct Document {
-    pub uri: String
+pub fn sk_from_def_kind(k: raw::DefKind) -> SymbolKind {
+    match k {
+        raw::DefKind::Enum => SymbolKind::Enum,
+        raw::DefKind::Tuple => SymbolKind::Array,
+        raw::DefKind::Struct => SymbolKind::Class,
+        raw::DefKind::Trait => SymbolKind::Interface,
+        raw::DefKind::Function => SymbolKind::Function,
+        raw::DefKind::Method => SymbolKind::Function,
+        raw::DefKind::Macro => SymbolKind::Function,
+        raw::DefKind::Mod => SymbolKind::Module,
+        raw::DefKind::Type => SymbolKind::Interface,
+        raw::DefKind::Local => SymbolKind::Variable,
+        raw::DefKind::Static => SymbolKind::Variable,
+        raw::DefKind::Const => SymbolKind::Variable,
+        raw::DefKind::Field => SymbolKind::Variable,
+        raw::DefKind::Import => SymbolKind::Module,
+    }
 }
 
-impl_file_name!(Document);
-
-#[allow(non_snake_case)]
-#[derive(Debug, Deserialize)]
-pub struct TextDocumentIdentifier {
-    pub uri: String
+pub fn new_completion_item(label: String, detail: String) -> CompletionItem {
```

Comment:
Could this have a more descriptive name? It is not really a CompletionItem ctor fn